File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < food.length; i++) {
            if (food[i] == 1) {
                continue;
            }
            sb.append(String.valueOf(i).repeat(Math.max(0, food[i] / 2)));
        }
        String[] split = sb.toString().split("");
        Arrays.sort(split, Comparator.reverseOrder());
        sb.append("0");
        sb.append(String.join("", split));
        return sb.toString();
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.swift
import Foundation

func solution(_ food:[Int]) -> String {
    var str: [String] = []

    for i in 1..<food.count {
        var n = food[i]
        if food[i] % 2 == 1 {
            n = food[i] - 1
        }
        for j in 0..<n/2 {
            str.append(String(i))
        }
    }
    str.append("0")

    for i in (1..<food.count).reversed() {
        var n = food[i]
        if food[i] % 2 == 1 {
            n = food[i] - 1
        }
        for j in 0..<n/2 {
            str.append(String(i))
        }
    }

    return str.joined()
}